inference-engine/tests/functional/plugin/shared/src/single_layer_tests/tensor_iterator.cpp

```diff
@@ -0,0 +1,230 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <memory>
+#include <functional>
+
+#include "ie_core.hpp"
+
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "functional_test_utils/precision_utils.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+
+#include "single_layer_tests/tensor_iterator.hpp"
+#include <transformations/control_flow/unroll_tensor_iterator.hpp>
+
+namespace LayerTestsDefinitions {
+
+    std::string TensorIteratorTest::getTestCaseName(const testing::TestParamInfo<TensorIteratorParams> &obj) {
+        bool should_decompose;
+        size_t seq_lenghts;
+        size_t batch;
+        size_t hidden_size;
+        size_t input_size;
+        TensorIteratorBody ti_body;
+        float clip;
+        ngraph::op::RecurrentSequenceDirection direction;
+        InferenceEngine::Precision netPrecision;
+        std::string targetDevice;
+        std::tie(should_decompose, seq_lenghts, batch, hidden_size, input_size, clip, ti_body, direction, netPrecision,
+                 targetDevice) = obj.param;
+        std::vector<std::vector<size_t>> inputShapes = {};
+
+        std::string type;
+        switch (ti_body) {
+            case TensorIteratorBody::LSTM:
+                type = "LSTM";
```

Comment:
We can move this logic to ngraph_helpers (Can be reused in the future, see overloaded operators in the end of file)

Reply:
It should be done after the creation of ngraph::function, I guess it a suitable place

Reply:
Not in this PR, have no time for it, not a blocker. Please create a separate ticket, if necessary

Reply:
Could you please create?

Reply:
43220